core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -646,7 +645,17 @@ public ParseField getParseField() {
             return parseField;
         }
 
-        public ScriptMode getDefaultScriptMode() {
+        public ScriptMode getDefaultScriptMode(@Nullable String language) {
+            if (language != null) {
```

Comment:
I agree with @nik9000 that it would be cleaner for each language/plugin to specify its own defaults.